### PR TITLE
feat: Responsive dashboard

### PR DESCRIPTION
**Pull request by sumit-158**
### What
- Dashboard 

### Screenshot
image

### CODE REVIEW

**Comment by g123k** [comment]
Please don't forget to provide 2x, 3x… pictures or use SVG please: https://docs.flutter.dev/development/ui/assets-and-images

File: client/lib/helpers/responsiveness.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class Responsive extends StatelessWidget {
+  final Widget mobile;
+  final Widget? tablet;
+  final Widget desktop;
+
+  const Responsive({
+    Key? key,
+    required this.mobile,
+    this.tablet,
```
Comment (g123k) [inline client/lib/helpers/responsiveness.dart:11]: Please ensure to always have required elements at the beginning of the constructor

File: client/lib/helpers/responsiveness.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  static bool isDesktop(BuildContext context) =>
+      MediaQuery.of(context).size.width >= 1100;
+
+  @override
+  Widget build(BuildContext context) {
+    final Size size = MediaQuery.of(context).size;
+    if (size.width >= 1100) {
+      return desktop;
+    } else if (size.width >= 850 && size.width < 1100) {
```
Comment (g123k) [inline client/lib/helpers/responsiveness.dart:30]: Where do those values come from?

File: client/lib/layout.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import 'package:client/helpers/responsiveness.dart';
+
+import 'package:client/widgets/dash_board_contents.dart';
+import 'package:client/widgets/side_menu.dart';
+
+import 'package:flutter/material.dart';
+
+import 'widgets/top_nav.dart';
```
Comment (g123k) [inline client/lib/layout.dart:8]: Please always use `package:client… imports`

File: client/lib/layout.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+class SiteLayout extends StatelessWidget {
+  final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey();
```
Comment (g123k) [inline client/lib/layout.dart:11]: Why do you need this key?

File: client/lib/layout.dart [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    return Scaffold(
+      key: scaffoldKey,
+      extendBodyBehindAppBar: true,
+      appBar: topNavigationBar(context, scaffoldKey),
+      drawer: const Drawer(
+        child: SideMenu(),
+      ),
+      body: const Responsive(
+          desktop: DashBoardContent(),
+          mobile: Padding(padding: EdgeInsets.symmetric(horizontal: 16))),
```
Comment (g123k) [inline client/lib/layout.dart:26]: Please add extra commas

File: client/lib/main.dart [hunk 6]
```diff
@@ -11,60 +13,10 @@ class MyApp extends StatelessWidget {
...
-        ),
-      ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: _incrementCounter,
-        tooltip: 'Increment',
-        child: const Icon(Icons.add),
-      ), // This trailing comma makes auto-formatting nicer for build methods.
+      debugShowCheckedModeBanner: false,
+      title: 'Dashboard',
+      theme: lightTheme,
```
Comment (g123k) [inline client/lib/main.dart:18]: Why do you force the light theme here?

File: client/lib/routing/roter.dart [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import 'package:client/routing/routes.dart';
```
Comment (g123k) [inline client/lib/routing/roter.dart:1]: The file is named "roter", don't you mean "router" instead?

File: client/lib/themes/theme_constants.dart [hunk 8]
```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+
+Color primary = const Color(0xFFFF8714);
```
Comment (g123k) [inline client/lib/themes/theme_constants.dart:3]: Please say instead `const Color primary = Color(xxx)`

File: client/lib/themes/theme_constants.dart [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+Color onprimary = const Color(0xFF000000);
+Color secondary = const Color(0xFFF5F5F5);
+Color onsecondary = const Color(0xFFFFFFFF);
+Color background = const Color(0xFFFFFFFF);
+const apppadding = 16.0;
```
Comment (g123k) [inline client/lib/themes/theme_constants.dart:8]: Please add the type + add an uppercase => appPadding

File: client/lib/themes/theme_constants.dart [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(20.0),
+        ),
+      ),
+      backgroundColor: MaterialStateProperty.all<Color>(primary),
+    ),
+  ),
+  inputDecorationTheme: InputDecorationTheme(
+    border: OutlineInputBorder(
+        borderRadius: BorderRadius.circular(15), borderSide: BorderSide.none),
```
Comment (g123k) [inline client/lib/themes/theme_constants.dart:28]: Extra comma needed

File: client/lib/themes/theme_constants.dart [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    filled: true,
+    fillColor: secondary.withOpacity(1.0),
```
Comment (g123k) [inline client/lib/themes/theme_constants.dart:30]: An opacity of 1.0 doesn't do anything, why do you need it?

File: client/lib/widgets/side_menu_items.dart [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import 'package:client/themes/theme_constants.dart';
+import 'package:flutter/material.dart';
+
+class SideMenuList extends StatelessWidget {
+  const SideMenuList(
+      {Key? key,
+      required this.title,
+      required this.images,
+      required this.onclicked})
```
Comment (g123k) [inline client/lib/widgets/side_menu_items.dart:9]: Extra comma please

File: client/lib/widgets/side_menu_items.dart [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+      : super(key: key);
+
+  final String title, images;
+  final VoidCallback onclicked;
```
Comment (g123k) [inline client/lib/widgets/side_menu_items.dart:13]: onclicked -> `onClicked`

File: client/lib/widgets/side_menu_items.dart [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+class SideMenuList extends StatelessWidget {
+  const SideMenuList(
+      {Key? key,
+      required this.title,
+      required this.images,
+      required this.onclicked})
+      : super(key: key);
+
+  final String title, images;
```
Comment (g123k) [inline client/lib/widgets/side_menu_items.dart:12]: Please separate each attribute (= one by line)

File: client/lib/widgets/top_nav.dart [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+              onPressed: () {
+                key.currentState!.openDrawer();
+              }),
+      title: Row(
+        children: [
+          Visibility(
+            visible: !Responsive.isMobile(context),
+            child: const Text('Open Food Facts'),
+          ),
+          Expanded(child: Container()),
```
Comment (g123k) [inline client/lib/widgets/top_nav.dart:31]: Why do you need this? A empty `Container` is really expensive

File: client/lib/widgets/top_nav.dart [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+          IconButton(
+              icon: const Icon(
+                Icons.search,
+                color: Colors.black,
+              ),
+              onPressed: () {}),
+          Container(
+            width: 1,
+            height: 22,
```
Comment (g123k) [inline client/lib/widgets/top_nav.dart:40]: PLease use a percentage here instead

File: client/pubspec.yaml [hunk 17]
```diff
@@ -1,26 +1,93 @@
 name: client
-description: Community portal frontend
-repository: https://github.com/openfoodfacts/community-portal
+description: A new Flutter project.
```
Comment (g123k) [inline client/pubspec.yaml:2]: Why did you change the description and URL?

File: client/pubspec.yaml [hunk 18]
```diff
@@ -1,26 +1,93 @@
...
     sdk: flutter
 
-  openfoodfacts_flutter_lints:
-    git: https://github.com/openfoodfacts/openfoodfacts_flutter_lints.git
+  # The "flutter_lints" package below contains a set of recommended lints to
+  # encourage good coding practices. The lint set provided by the package is
+  # activated in the `analysis_options.yaml` file located at the root of your
+  # package. See that file for information about deactivating specific lint
+  # rules and activating additional ones.
+  flutter_lints: ^2.0.0
```
Comment (g123k) [inline client/pubspec.yaml:51]: No, no, no, please use the shared lints

**Review by g123k — CHANGES_REQUESTED** [review]

**Comment by g123k** [comment]
> Hey, @g123k can you please help me, I'm having hard time to make local navigation I have done something can you tell where I am going wrong.

Your navigation seems OK. It's just you have to pass to the MaterialApp your router

File: client/lib/helpers/responsiveness.dart (hunk 2 above)
Comment (sumit-158) [reply client/lib/helpers/responsiveness.dart:30]: [here ](https://medium.com/flutter-community/making-cross-platform-flutter-landing-page-responsive-7fffe0655970)

File: client/lib/main.dart (hunk 6 above)
Comment (sumit-158) [reply client/lib/main.dart:18]: AppBar theme is inside lighttheme

File: client/lib/routing/roter.dart (hunk 7 above)
Comment (sumit-158) [reply client/lib/routing/roter.dart:1]: my bad! I don't why I do such silly mistakes

File: client/lib/helpers/responsiveness.dart (hunk 2 above)
Comment (g123k) [reply client/lib/helpers/responsiveness.dart:30]: Ok, it would be better if you rely on an external library:

https://pub.dev/packages/responsive_builder
https://pub.dev/packages/responsive_sizer
https://pub.dev/packages/adaptive_breakpoints
…

File: client/lib/layout.dart (hunk 4 above)
Comment (g123k) [reply client/lib/layout.dart:11]: Ok, then you only need this keep in your Widget, not here

File: client/lib/themes/theme_constants.dart (hunk 9 above)
Comment (sumit-158) [reply client/lib/themes/theme_constants.dart:8]: ??

File: client/lib/main.dart (hunk 6 above)
Comment (g123k) [reply client/lib/main.dart:18]: Ok then create a duplicate one with the dark mode, please

File: client/lib/themes/theme_constants.dart (hunk 11 above)
Comment (sumit-158) [reply client/lib/themes/theme_constants.dart:30]: solved

Comment (g123k) [reply client/lib/themes/theme_constants.dart:30]: Then please commit, it please 😀 (don't mark the conversation as resolved until)

File: client/lib/widgets/top_nav.dart (hunk 15 above)
Comment (sumit-158) [reply client/lib/widgets/top_nav.dart:31]: image

File: client/pubspec.yaml (hunk 17 above)
Comment (sumit-158) [reply client/pubspec.yaml:2]: By mistake it got pushed.

File: community_portal/api/admin.py [hunk 19]
```diff
@@ -1,3 +1,6 @@
-from django.contrib import admin
```
[the remark this answers is not in the document]
Comment (sumit-158) [reply community_portal/api/admin.py:1]: actually I pushed both in same branch by mistake

[the remark this answers is not in the document]
Comment (sumit-158) [reply community_portal/api/admin.py:1]: I will revert that changes

File: client/lib/helpers/responsiveness.dart (hunk 2 above)
Comment (sumit-158) [reply client/lib/helpers/responsiveness.dart:30]: Wow! that more easier than my approach , thank you @g123k
